src/Storage/Storage.Management/File/GetAzureStorageShare.cs

```diff
@@ -0,0 +1,124 @@
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Storage Account Name.",
+            ParameterSetName = AccountNameParameterSet)]
+        [Alias(AccountNameAlias)]
+        [ValidateNotNullOrEmpty]
+        public string StorageAccountName { get; set; }
+
+        [Parameter(Mandatory = true,
+            HelpMessage = "Storage account object",
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = AccountObjectParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public PSStorageAccount StorageAccount { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Input a File Share Resource Id.",
+           ParameterSetName = ShareResourceIdParameterSet)]
+        [ValidateNotNullOrEmpty]
+        public string ResourceId { get; set; }
+
+        [Alias("N", "ShareName")]
+        [Parameter(HelpMessage = "Share Name",
+            Mandatory = false,
+            ValueFromPipeline = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string Name { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            base.ExecuteCmdlet();
+
+            switch (ParameterSetName)
+            {
+                case AccountObjectParameterSet:
+                    this.ResourceGroupName = StorageAccount.ResourceGroupName;
+                    this.StorageAccountName = StorageAccount.StorageAccountName;
+                    break;
+                case ShareResourceIdParameterSet:
+                    ResourceIdentifier shareResource = new ResourceIdentifier(ResourceId);
+                    this.ResourceGroupName = shareResource.ResourceGroupName;
+                    this.StorageAccountName = PSBlobServiceProperties.GetStorageAccountNameFromResourceId(ResourceId);
+                    this.Name = shareResource.ResourceName;
+                    break;
+                default:
+                    // For AccountNameParameterSet, the ResourceGroupName and StorageAccountName can get from input directly
+                    break;
+            }
+
+            if (!string.IsNullOrEmpty(this.Name))
+            {
+                var Share = this.StorageClient.FileShares.Get(
+                           this.ResourceGroupName,
+                           this.StorageAccountName,
+                           this.Name);
+                WriteObject(new PSShare(Share));
+            }
+            else
+            {
+                var Shares = this.StorageClient.FileShares.List(
+                           this.ResourceGroupName,
+                           this.StorageAccountName);
+                WriteShareList(Shares);
```

Comment:
Will this return PSShare, or somethign else?

Reply:
This will write a list of PSShare.

See the code of the function:
```c#
            if (shares != null)
            {
                List<PSShare> output = new List<PSShare>();
                shares.ForEach(share => output.Add(new PSShare(share)));
                WriteObject(output, true);
            }
```